### ISSUE TITLE
Accumulate iterator adapter for rust itertools (like itertools.accumulate in Python itertools)

### ISSUE THREAD

**Issue by evan0greenup** [rust-itertools/itertools#705]
I don't find iterator adapter like [`itertools.accumulate(iterable[, func, *, initial=None])`](https://docs.python.org/3/library/itertools.html#itertools.accumulate) in Rust itertools.

It would be nice to add it in.

**Comment by phimuemue** [comment]
Isn't this [`scan`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.scan)?

**Comment by Philippe-Cholet** [comment]
Yes it very similar to `scan` but it's not very obvious at first look. With `initial=0` and the default `func` the addition.

    (1..8).scan(0, |x, y| { *x += y; Some(*x) })
while people from coming Python would expect something more like

    (1..8).some_name(0, |x, y| x + y)
Like in #147
